### PR TITLE
[Kube-Gen] Deploy.sh 

### PR DESCRIPTION
**Pull request by Lewiky**
Adds generation for the deploy.sh file we used to run everything through kube. Pretty constant apart from adding one extra line per service, and a conditional section added when metrics are used.

### CODE REVIEW

File: src/main/resources/shell/deploy.sh.header.snippet [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+minikube start --vm-driver=virtualbox
+
+echo $GREEN
+
+kubectl create secret docker-registry regcred --docker-server=$REG_URL --docker-username=$REG_USERNAME --docker-password=$REG_PASSWORD --docker-email=$REG_EMAIL
+
+echo
+echo Creating ConfigMaps...
+echo
```
Comment (c-harding) [inline src/main/resources/shell/deploy.sh.header.snippet]: newline plz

File: src/main/resources/shell/deploy.sh.metrics.snippet [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+# Grafana
+kubectl create configmap grafana-datasource-config --from-file "$BASEDIR/grafana/provisioning/datasources/datasource.yml" -o=yaml
+
+for file in "$BASEDIR/grafana/provisioning/dashboards/"*
+do
+  filename=$(basename $file) # Get everything after the final /
+  file=$(echo $filename | cut -f 1 -d '.') # Get everything before the . (filename without extension)
+  kubectl create configmap "grafana-$file-config" --from-file "$BASEDIR/grafana/provisioning/dashboards/$filename" -o=yaml
+done
```
Comment (c-harding) [inline src/main/resources/shell/deploy.sh.metrics.snippet]: newline plz

File: src/test/scala/temple/TestUtils.scala [hunk 3]
```diff
@@ -24,7 +24,10 @@ object TestUtils {
           FileVisitResult.CONTINUE
 
         override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
-          foundFiles += File(path.relativize(file).getParent.toString, file.getFileName.toString) -> FileUtils.readFile(
+          foundFiles += File(
+            Option(path.relativize(file).getParent).map(_.toString).getOrElse(""),
```
Comment (c-harding) [inline src/test/scala/temple/TestUtils.scala:28]: When does this return `null`?

**Review by c-harding — APPROVED** [review]
nice

File: src/test/scala/temple/TestUtils.scala (hunk 3 above)
Comment (Lewiky) [reply src/test/scala/temple/TestUtils.scala:28]: When the `file` you're relativizing to is in the same directory as `path`,  `getParent` is `null`

Comment (jaylees14) [reply src/test/scala/temple/TestUtils.scala:28]: Nice that 😉

**Review by jaylees14 — APPROVED** [review]
Looks good, think you need to merge

**Pull request merged**